File: pipes/wc.js
const { net, sys, print, error } = just
const { read } = net
const { exit, strerror, errno, STDIN_FILENO } = sys
const BUFSIZE = 65536

const buf = new ArrayBuffer(BUFSIZE)
let size = 0
let n = 0
while ((n = read(STDIN_FILENO, buf))) size += n
if (n < 0) {
  error(`read: ${strerror(errno())} (${errno()})`)
  exit(1)
}
print(`size: ${size}`)

File: pipes/wc.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <errno.h>
#include <string.h>

#define BUFSIZE 65536

int main(int argc, char *argv[]) {
  char buf[BUFSIZE];
  unsigned long size = 0;
  int n = 0;
  while (n = read(STDIN_FILENO, buf, BUFSIZE)) size += n;
  if (n < 0) {
    fprintf(stderr, "read: %s (%i)\n", strerror(errno), errno);
    exit(1);
  }
  fprintf(stdout, "size: %lu\n", size);
}
